src/Lykke.RabbitMq.Azure/DeferredMessagesRepository.cs

```diff
@@ -0,0 +1,197 @@
+        /// </p>
+        /// <p>
+        /// If your publishes about 70 messages per minute and in the average the message is deferred about 1 hour, 
+        /// <see cref="DeferredMessagesPartitionSize.Minute"/> is a good choice. Because a single partition will 
+        /// contains about 70 messages, and there will be about 60 partitions of not overdue messages in the storage
+        /// </p>
+        /// Default value is <see cref="DeferredMessagesPartitionSize.Minute"/>
+        /// </param>
+        public static IDeferredMessagesRepository Create(
+            ILog log,
+            IReloadingManager<string> connectionString,
+            string tableName, DeferredMessagesPartitionSize partitionSize = DeferredMessagesPartitionSize.Minute)
+        {
+            return new DeferredMessagesRepository(
+                AzureTableStorage<DeferredMessageEntity>.Create(connectionString, tableName, log),
+                partitionSize);
+        }
+
+        /// <summary>
+        /// Stores deferred messages in the Azure Table Storage.
+        /// Note, that message size is limited to the 65 Kb 
+        /// and you can publish only up to the 1000 messages for the same delivery moment
+        /// </summary>
+        /// <param name="storage">The storage</param>
+        /// <param name="partitionSize">
+        /// <p>
+        /// Messages partition size in time.
+        /// Choose actual value accroding to your typical deferred messages delay period and the messages rate.
+        /// Good value is when amount of partitions is balanced with amount of the messages in the single partition.
+        /// </p>
+        /// <p>
+        /// If your publishes about 70 messages per minute and in the average the message is deferred about 1 hour, 
+        /// <see cref="DeferredMessagesPartitionSize.Minute"/> is a good choice. Because a single partition will 
+        /// contains about 70 messages, and there will be about 60 partitions of not overdue messages in the storage
+        /// </p>
+        /// Default value is <see cref="DeferredMessagesPartitionSize.Minute"/>
+        /// </param>
+        public DeferredMessagesRepository(
+            INoSQLTableStorage<DeferredMessageEntity> storage, 
+            DeferredMessagesPartitionSize partitionSize = DeferredMessagesPartitionSize.Minute)
+        {
+            _storage = storage ?? throw new ArgumentNullException(nameof(storage));
+            _partitionSize = partitionSize;
+        }
+
+        /// <inheritdoc cref="IDeferredMessagesRepository.SaveAsync"/>
+        public async Task SaveAsync(byte[] message, DateTime deliverAt)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            var pk = GetPartitionKey(deliverAt);
+            var rk = GetRowKey(deliverAt, Interlocked.Increment(ref _counter));
```

Comment:
When _counter should be reset to 0?

Reply:
only after app restart

Reply:
Let's assume, that message rate is 1000 per second. Since counter is `long`, to overflow it app should work for about 300 million years without restarts.